engine/is_raspberrypi.py

```diff
@@ -0,0 +1,12 @@
+import io
+import os
+"""a function from : https://raspberrypi.stackexchange.com/questions/5100/detect-that-a-python-program-is-running-on-the-pi/74541#74541"""
+def is_raspberrypi():
+    try:
+        with io.open('/sys/firmware/devicetree/base/model', 'r') as m:
+            if 'raspberry pi' in m.read().lower(): return True
+    except Exception: pass
+    return False
+
+
+print(os.uname())
```

Comment:
Let's take this file out of this PR and put it on a new branch (since checking whether you're on a raspberry pi is a diff task).